### PR TITLE
feat: add `UserContextMenuInteraction` and `MessageContextMenuInteraction`

### PR DESCRIPTION
**Pull request by GrapeColor**
**Please describe the changes this PR makes and why it should be merged:**
Add `UserContextMenuInteraction` and `MessageContextMenuInteraction`.
Add `targetUser` and `targetMember` properties in `UserContextMenuInteraction`.
Add a `targetMessage` property in `MessageContextMenuInteraction`.

Currently, the context menu interaction is unified to `ContextMenuInteraction`.
Therefore, target users and messages must use `interaction.options.getUser('user')` and `interaction.options.getMessage('message')`.
This isn't intuitive, so we'll separate the classes and add getter functions.

**Status and versioning classification:**
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by suneettipirneni — COMMENTED** [review]
I might be missing something here, but where are these new classes being created in the lib? The typeguard is misleading if the actual runtime version of the class is still a `ContextMenuInteraction` and not a more specific type.

File: src/structures/UserContextMenuInteraction.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    return this.options.getUser('user');
+  }
+
+  /**
+   * The member this interaction was sent from
+   * @type {?(GuildMember|APIGuildMember)}
+   * @readonly
+   */
+  get targetMember() {
+    return this.options.getMember(this.targetId);
```
Comment (ImRodry) [inline src/structures/UserContextMenuInteraction.js:25]:
```suggestion
    return this.options.getMember(‘user’);
```

Comment (almeidx) [reply src/structures/UserContextMenuInteraction.js:25]:
```js
    return this.options.getMember('user');
```

Comment (GrapeColor) [reply src/structures/UserContextMenuInteraction.js]: That's right.
Thank you.

**Comment by GrapeColor** [comment]
> I might be missing something here, but where are these new classes being created in the lib? The typeguard is misleading if the actual runtime version of the class is still a `ContextMenuInteraction` and not a more specific type.

I'm sorry.
Now I have added a new class to the `InteractionCreate`.

File: typings/index.d.ts [hunk 2]
```diff
@@ -1526,6 +1528,13 @@ export class MessageComponentInteraction<Cached extends CacheType = CacheType> e
   public static resolveType(type: MessageComponentTypeResolvable): MessageComponentType;
 }
 
+export class MessageContextMenuInteracion<Cached extends CacheType = CacheType> extends ContextMenuInteraction<Cached> {
```
Comment (suneettipirneni) [inline typings/index.d.ts:1531]: typo
```suggestion
export class MessageContextMenuInteraction<Cached extends CacheType = CacheType> extends ContextMenuInteraction<Cached> {
```

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

File: typings/index.d.ts [hunk 3]
```diff
@@ -1526,6 +1528,13 @@ export class MessageComponentInteraction<Cached extends CacheType = CacheType> e
   public static resolveType(type: MessageComponentTypeResolvable): MessageComponentType;
 }
 
+export class MessageContextMenuInteraction<Cached extends CacheType = CacheType> extends ContextMenuInteraction<Cached> {
+  public targetMessage: CacheTypeReducer<Cached, Message, APIMessage>;
```
Comment (vladfrangu) [inline typings/index.d.ts:1532]:
```suggestion
  public readonly targetMessage: CacheTypeReducer<Cached, Message, APIMessage>;
```

File: typings/index.d.ts [hunk 4]
```diff
@@ -2169,6 +2178,14 @@ export class User extends PartialTextBasedChannel(Base) {
   public toString(): UserMention;
 }
 
+export class UserContextMenuInteraction<Cached extends CacheType = CacheType> extends ContextMenuInteraction<Cached> {
+  public targetUser: User;
+  public targetMember: CacheTypeReducer<Cached, GuildMember, APIInteractionGuildMember>;
```
Comment (vladfrangu) [inline typings/index.d.ts:2183]:
```suggestion
  public readonly targetUser: User;
  public readonly targetMember: CacheTypeReducer<Cached, GuildMember, APIInteractionGuildMember>;
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by GrapeColor** [comment]
Thank you all for your helps.

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by GrapeColor** [comment]
I'm sorry for your review.
I forgot to add the classes to `index.js`.
Really sorry.

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**